include/ruby/debug.h

```diff
@@ -595,6 +607,31 @@ RBIMPL_ATTR_NONNULL(())
  */
 VALUE rb_tracearg_raised_exception(rb_trace_arg_t *trace_arg);
 
+RBIMPL_ATTR_NONNULL(())
+/**
+ * Queries the compiled source code of the 'script_compiled' event.
+ * If loaded from a file, it will return nil.
+ *
+ * @param[in]  trace_arg         A trace instance
+ * @exception  rb_eRuntimeError  The tracing event is not 'script_compiled'.
+ * @retval     RUBY_Qnil         The script was loaded from a file.
+ * @retval     otherwise         The compiled source code.
+ *
+ */
+VALUE rb_tracearg_eval_script(rb_trace_arg_t *trace_arg);
+
+RBIMPL_ATTR_NONNULL(())
+/**
+ *
+ * Queries the compiled instruction sequence on a 'script_compiled' event.
```

Comment:
Could you add it is MRI specific?
https://ruby-doc.org/core-2.6/TracePoint.html#method-i-instruction_sequence

Reply:
I added the note from the docs you mentioned. Thanks for finding this!